src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -3821,6 +3821,82 @@ internal new NamedTypeSymbol CreateNativeIntegerTypeSymbol(bool signed)
             return this.AnonymousTypeManager.ConstructAnonymousTypeSymbol(descriptor).GetPublicSymbol();
         }
 
+        protected override IMethodSymbol CommonCreateBuiltinOperator(
+            string name,
+            ITypeSymbol returnType,
+            ITypeSymbol leftType,
+            ITypeSymbol rightType,
+            bool isChecked)
+        {
+            var csharpReturnType = returnType.EnsureCSharpSymbolOrNull(nameof(returnType));
+            var csharpLeftType = leftType.EnsureCSharpSymbolOrNull(nameof(leftType));
+            var csharpRightType = rightType.EnsureCSharpSymbolOrNull(nameof(rightType));
+
+            // caller already checked all of these were not null.
+            Debug.Assert(csharpReturnType is not null);
+            Debug.Assert(csharpLeftType is not null);
+            Debug.Assert(csharpRightType is not null);
+
+            var nameOk = name
+                is WellKnownMemberNames.CheckedAdditionOperatorName
+                or WellKnownMemberNames.AdditionOperatorName
+                or WellKnownMemberNames.BitwiseAndOperatorName
+                or WellKnownMemberNames.CheckedDivisionOperatorName
+                or WellKnownMemberNames.DivisionOperatorName
+                or WellKnownMemberNames.EqualityOperatorName
+                or WellKnownMemberNames.GreaterThanOperatorName
+                or WellKnownMemberNames.GreaterThanOrEqualOperatorName
+                or WellKnownMemberNames.LeftShiftOperatorName
+                or WellKnownMemberNames.LessThanOperatorName
+                or WellKnownMemberNames.LessThanOrEqualOperatorName
+                or WellKnownMemberNames.CheckedMultiplyOperatorName
+                or WellKnownMemberNames.MultiplyOperatorName
+                or WellKnownMemberNames.BitwiseOrOperatorName
+                or WellKnownMemberNames.InequalityOperatorName
+                or WellKnownMemberNames.ModulusOperatorName
+                or WellKnownMemberNames.RightShiftOperatorName
+                or WellKnownMemberNames.UnsignedRightShiftOperatorName
+                or WellKnownMemberNames.CheckedSubtractionOperatorName
+                or WellKnownMemberNames.SubtractionOperatorName
+                or WellKnownMemberNames.ExclusiveOrOperatorName;
```

Comment:
Does this work?

```suggestion
            var nameOk = SyntaxFacts.GetOperatorKind(name) != SyntaxKind.None;
```

Reply:
My suggestion isn't correct as we want to handle binary operators only in this path.

My concern is that we can forget updating this public API if new operators are added in future. It would be great if we can have a Debug assert in `GetOperatorKind` that one of the two APIs handle the new operator.

Reply:
no, that would support a superset of actually supported operator names.

Reply:
@CyrusNajmabadi Yup, already noticed that and commented. 

> My concern is that we can forget updating this public API if new operators are added in future. It would be great if we can have a Debug assert in `GetOperatorKind` that one of the two APIs handle the new operator.

This is still concerning, I think.

Reply:
Do you have a suggestion on what this assert would do?  It's unclear to me that GetOperatorKind will always map to something one of these can create.

Reply:
> Do you have a suggestion on what this assert would do? It's unclear to me that GetOperatorKind will always map to something one of these can create.

Maybe break the two `nameOk` checks into a two helpers (`IsBinaryOperatorName` and `IsUnaryOperatorName`) and for "none `SyntaxKind.None`" returns, we assert that one of those two helpers return true. Not sure if that makes sense

Reply:
> break the two nameOk checks into a two helpers

alternatively, we could have a "fake debug only compilation" that we call the APIs on for the purpose of the assert.